core/store/src/adapter/trie_store.rs

```diff
@@ -188,30 +198,21 @@ impl<'a> TrieStoreUpdateAdapter<'a> {
 /// It is kept out of `TrieStoreAdapter`, so that `TrieStoreUpdateAdapter` can use it without
 /// cloning `store` each time, see https://github.com/near/nearcore/pull/12232#discussion_r1804810508.
 pub fn get_shard_uid_mapping(store: &Store, child_shard_uid: ShardUId) -> ShardUId {
+    maybe_get_shard_uid_mapping(store, child_shard_uid).unwrap_or(child_shard_uid)
+}
+
+/// Get the `ShardUId` mapping for child_shard_uid. If the mapping does not exist, return None.
+fn maybe_get_shard_uid_mapping(store: &Store, child_shard_uid: ShardUId) -> Option<ShardUId> {
     store
         .get_ser::<ShardUId>(DBCol::StateShardUIdMapping, &child_shard_uid.to_bytes())
         .unwrap_or_else(|_| {
             panic!("get_shard_uid_mapping() failed for child_shard_uid = {}", child_shard_uid)
         })
-        .unwrap_or(child_shard_uid)
 }
 
-/// Constructs db key to be used to access the State column.
-/// First, it consults the `StateShardUIdMapping` column to map the `shard_uid` prefix
-/// to its ancestor in the resharding tree (according to Resharding V3)
-/// or map to itself if the mapping does not exist.
-///
-/// Please note that the mapped shard uid is read from db each time which may seem slow.
-/// In practice the `StateShardUIdMapping` is very small and should always be stored in the RocksDB cache.
-/// The deserialization of ShardUId is also very cheap.
-fn get_key_from_shard_uid_and_hash(
-    store: &Store,
-    shard_uid: ShardUId,
-    hash: &CryptoHash,
-) -> [u8; 40] {
-    let mapped_shard_uid = get_shard_uid_mapping(store, shard_uid);
```

Comment:
Having this line here was really nice -> can we keep it here and have this function return two values? Or some nice [iterable] struct so that it's type safe and convenient to use?

Reply:
I thought a bit about this while implementing and even tried it out, returning an iterator, but figured since this is an internal only function, it shouldn't matter too much and ease of view/less hopping around to see what code is being run is a better strategy here.